src/components/dialogs/dialogAlert/dialogAlertRoot/dialogAlertRoot.tsx

```diff
@@ -0,0 +1,83 @@
+import { Content, Overlay, Portal, Root } from '@radix-ui/react-alert-dialog';
+import classNames from 'classnames';
+import type React from 'react';
+import { createContext, useMemo, type ReactNode } from 'react';
+
+export type DialogAlertVariant = 'critical' | 'info' | 'success' | 'warning';
+
+export interface IDialogAlertRootProps {
+    children?: ReactNode;
+    /**
+     * Additional CSS class names for custom styling of the dialog's content container.
+     */
+    containerClassName?: string;
+    /**
+     * Manages the visibility state of the dialog. Should be implemented alongside `onOpenChange` for controlled usage.
+     */
+    open?: boolean;
+    /**
+     * Additional CSS class names for custom styling of the overlay behind the dialog.
+     */
+    overlayClassName?: string;
+    /**
+     * The visual style variant of the dialog.
+     * @default info
+     */
+    variant?: DialogAlertVariant;
+    /**
+     * Callback function invoked when the open state of the dialog changes.
+     */
+    onOpenChange?(open: boolean): void;
+    /**
+     * Handler called when focus moves to the trigger after closing
+     */
+    onCloseAutoFocus?: (e: Event) => void;
+    /**
+     * Handler called when focus moves to the destructive action after opening
+     */
+    onOpenAutoFocus?: (e: Event) => void;
+}
+
+export interface IDialogAlertContext {
+    variant: DialogAlertVariant;
+}
+
+export const DialogAlertContext = createContext<IDialogAlertContext>({ variant: 'info' });
+
+/**
+ * `DialogAlert.Root` component.
+ */
+export const DialogAlertRoot: React.FC<IDialogAlertRootProps> = (props) => {
```

Comment:
Currently the dialog does not close automatically on esc keydown, is this intended?

Reply:
Ah, yes. I didn't catch that. Because the dialog is controlled, this has to be handled manually. Good catch